packages/config/lib/options.ts

```diff
@@ -163,26 +185,43 @@ const resolvedOptions: Array<ResolvedConfigOption> = [
     name: 'includeShadowDom',
     defaultValue: false,
     validation: validate.isBoolean,
+    isWriteable: true,
   }, {
     name: 'integrationFolder',
     defaultValue: 'cypress/integration',
     validation: validate.isString,
     isFolder: true,
-  }, {
+  },
+  {
+    name: 'isInteractive',
+    defaultValue: true,
+    validation: validate.isBoolean,
+    isWriteable: true,
```

Comment:
Hmmm... this is tricky. We write this one internally and for our own tests since we're testing Cypress itself, but I don't think it's a good idea to make it publicly writeable.

Reply:
Right...we might have to add a way to distinguish between internal use because this situation actually comes up a few times

Reply:
Maybe skip writeable validation altogether if `CYPRESS_INTERNAL_ENV === 'development'` or maybe we need some other env variable for this.

Reply:
@chrisbreiding Would `isInteractive` be considered internal or just read-only? Also should it be included as a RuntimeConfigOption rather than a ResolvedConfigOption?

Reply:
It's [documented](https://docs.cypress.io/guides/references/configuration#isInteractive), so it should only be ready-only, not internal. I believe it should be a RuntimeConfigOption. I think ResolvedConfigOption is for options that can be specified in cypress.json.